R/calcHDDCDD.R

```diff
@@ -1,101 +1,714 @@
+      frsds <- f[f$variable == "rsds"    & f$gcm == m,][[n, "file"]]
+      fsfc  <- f[f$variable == "sfcwind" & f$gcm == m,][[n, "file"]]
+      fhuss <- f[f$variable == "huss"    & f$gcm == m,][[n, "file"]]
+    }
+
+    print(paste("Processing temperature file:", ftas))
+
+    if(bait) {
+      hddcddCell <- calcStackHDDCDD(ftas,
+                                    t_lim,
+                                    countries,
+                                    pop,
+                                    hddcddFactor,
+                                    bait,
+                                    frsds = frsds,
+                                    fsfc  = fsfc,
+                                    fhuss = fhuss,
+                                    wBAIT = wBAIT,
+                                    params = params)}
+
+    else {
+      hddcddCell <- calcStackHDDCDD(ftas,
+                                    t_lim,
+                                    countries,
+                                    pop,
+                                    hddcddFactor,
+                                    bait)}
+    return(hddcddCell)
+  }
+
+
+  plotRaster <- function(r) {
+    library("ggplot2")
+    r_map <- stack(as.data.frame(raster::getValues(r)))
+    coords <- raster::xyFromCell(r, seq_len(raster::ncell(r)))
+    names(r_map) <- c('value', 'variable')
+    r_map <- cbind(coords, r_map)
+    ggplot(r_map) +
+      geom_tile(aes(x, y, fill = value)) +
+      facet_wrap(~ variable) +
+      scale_fill_gradientn(colours = rev(terrain.colors(225))) +
+      coord_equal()
+  }
+
+
+
+  # PARAMETERS------------------------------------------------------------------
+
+  terra::setGDALconfig(c("BIGTIFF = YES"))
+
+
+  # threshold temperature for heating and cooling [C]
+  # NOTE: Staffel gives global average of T_heat = 14, T_cool = 20
+  # t_lim <- list("HDD" = seq(12, 18), "CDD" = seq(20, 26))
+  t_lim <- list("HDD" = seq(14), "CDD" = seq(22))
+
+  # standard deviations for temperature distributions
+  tlim_std <- 5   # threshold
+  tamb_std <- 5   # ambient
+
+  # historical years    -> NOTE: not sure if further needed
+  firstHistYear <- 1950
+  lastHistYear  <- 2009
+
+  # range of pre-calculated HDD/CDD-values, e.g. [223, 232] K, converted to [C]
+  tlow <- 223 - 273.15
+  tup <- 323 - 273.15
+
+
+
+  #--- BAIT parameters
+
+  # The weights (x,y,z) for calcBAIT and the smoothing coefficient are assumed to
+  # be region-independent and equal to the mean of the values given in Staffell
+  # et al. 2023
+  x <- 0.012
+  y <- -0.20
+  z <- 0.05
+
+  # smoothing coefficient
+  sig <- 0.50
+
+  # The blending parameters for the blending of BAIT and raw temperature are like-wise
+  # taken from the paper.
+  bLower <- 15
+  bUpper <- 23
+  bMax   <- 0.5
+
+  # concatenate to vector
+  wBAIT <- c(x, y, z, sig, bLower, bUpper, bMax)
+
+  # BAIT parameter names
+  parNames <- c("a_rsds", "b_rsds", "a_sfcwind", "b_sfcwind", "a_huss", "b_huss")
```

Comment:
How about a named list? Then, one can interpret the values when being called in the code above.